Doc/whatsnew/3.12.rst

```diff
@@ -1065,6 +1065,18 @@ Deprecated
   contain the creation time, which is also available in the new ``st_birthtime``
   field. (Contributed by Steve Dower in :gh:`99726`.)
 
+* :mod:`os`: On POSIX platforms, :func:`os.fork` can now raise a
+  :exc:`DeprecationWarning` when it can detect being called from a
+  multithreaded process. There has always been a fundamental incompatibility
+  with the POSIX platform when doing so. Even if such code *appeared* to work.
+  We added the warning to to raise awareness as issues encounted by code doing
+  this are becoming more frequent. See the :func:`os.fork` documentation for
+  more details.
+
```

Comment:
I would copy the paragraph from the os.fork() docs here as well, to make it easier for people to find the reasoning behind this:

```
See `this discussion on fork being incompatible with threads
<https://discuss.python.org/t/33555>`_
for technical details of why we're surfacing this longstanding
platform compatibility problem to developers.
```

Reply:
Thanks, I included that (edited) over in the 3.12 backport https://github.com/python/cpython/pull/109773. I'll make the same edit in `main` after that is merged.